### DISCUSSION TITLE
How to update?

### DISCUSSION THREAD

**Discussion by dynabler** [jhuckaby/Cronicle#748] [Q&A] [answered]
Maybe it's trivial, but how do I update? 

Is updating the same as [installing](https://github.com/jhuckaby/Cronicle/blob/master/docs/Setup.md#installation) _without_ running [setup](https://github.com/jhuckaby/Cronicle/blob/master/docs/Setup.md#setup)? 

Perhaps it's an idea to mention it in the docs. Maybe someday special instructions on updating is required (breaking changes for example).

**Comment by jhuckaby** [comment, marked as answer]
https://github.com/jhuckaby/Cronicle/blob/master/docs/CommandLine.md#upgrading-cronicle

**Reply by dynabler** [reply]
well... you really have thought of everything ;-)

**Comment by strefethen** [comment]
First, this is such a great tool and I'm glad I found it.

I'm using Cronicle on an Amazon Lightsail instance which I documented here:

https://www.stevetrefethen.com/running-cronicle-on-an-amazon-lightsail-instance/

As a result my install is not using root so the upgrade command doesn't work as it requires root. Is there a path to manually installing the latest update?

**Reply by jhuckaby** [reply]
You can just follow the installation instructions again.  But stop Cronicle first, then start it again at the end.  Skip the "control.sh setup" step, as you only need to do that on initial install.

```sh
$ cd /opt/bitnami/apps/cronicle
$ ./bin/control.sh stop
$ curl -L https://github.com/jhuckaby/Cronicle/archive/master.tar.gz | tar zxvf - --strip-components 1
$ npm install
$ ./bin/control.sh start
```

This will preserve your config file.

**NOTE::* Your blog article says to download Cronicle v1.0.0.  There is no v1.0.0.  Might want to replace that with the latest, shown above.
